### ISSUE TITLE
Crash on Import Unreal

### ISSUE THREAD

**Issue by danouche93** [NegInfinity/ProjectExodus#8]
Hey,
I did a new topic because I have now a problem with Unreal.
When I try to Import the .json file to Unreal, it crashes and I get this error :

LoginId:f72b7e044e602f24b32ffea98818406c
EpicAccountId:113b31dff299415bac6058a219062d14

Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:D:\Build\++UE4\Sync\Engine\Source\Runtime\Core\Public\Containers/Array.h] [Line: 611] Array index out of bounds: 0 from an array of size 0

KERNELBASE
UE4Editor_Core
UE4Editor_Core
UE4Editor_Core
UE4Editor_MeshDescriptionOperations
UE4Editor_MeshBuilder
UE4Editor_MeshBuilder
UE4Editor_MeshBuilder
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_ExodusImport!MeshBuilder::setupStaticMesh() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\meshbuilder\staticmesh.cpp:163]
UE4Editor_ExodusImport!std::_Func_impl_no_alloc<<lambda_7caa6d6cf6c48161a0e26702bb774e93>,void,UStaticMesh *>::_Do_call() [c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.16.27023\include\functional:15732480]
UE4Editor_ExodusImport!<lambda_b8cc1043bbcdf4592ddd5391af60cb9d>::operator()() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\unrealutilities.h:141]
UE4Editor_ExodusImport!UnrealUtilities::createAssetPackage() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\unrealutilities.cpp:19]
UE4Editor_ExodusImport!UnrealUtilities::createAssetObject<UStaticMesh>() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\unrealutilities.h:146]
UE4Editor_ExodusImport!JsonImporter::importMesh() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\jsonimporter\mesh.cpp:125]
UE4Editor_ExodusImport!JsonImporter::loadMeshes() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\jsonimporter.cpp:171]
UE4Editor_ExodusImport!JsonImporter::importProject() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\jsonimporter\scene.cpp:149]
UE4Editor_ExodusImport!FJsonImportModule::PluginImportButtonClicked() [e:\shenkar\unreal engine\myproject\plugins\exodusimport\source\exodusimport\private\jsonimport.cpp:143]
UE4Editor_ExodusImport!TBaseRawMethodDelegateInstance<0,FJsonImportModule,void __cdecl(void)>::ExecuteIfSafe() [e:\ue_4.21\engine\source\runtime\core\public\delegates\delegateinstancesimpl.h:518]
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_Slate
UE4Editor_ApplicationCore
UE4Editor_ApplicationCore
UE4Editor_ApplicationCore
UE4Editor_ApplicationCore
user32
user32
UE4Editor_ApplicationCore
UE4Editor
UE4Editor
UE4Editor
UE4Editor
UE4Editor
kernel32
ntdll

I tried remove all the assets I had in the scene, also the colliders, but it still happens.

**Comment by NegInfinity** [comment]
Hi.

It looks like you hit a bug. I need a test data that causes that error, preferably something that is not copyrighted.

If you could either send me the exported files via email, or upload them somewhere, that would be great.

Hard to proceed without that.

Alternativley, let me know what asset have you tried to export and whether this happens when you work with a plain scene with basic shapes.

For example, take a look at installation videos, they depict basic test scene:

https://www.youtube.com/watch?v=z6Iy0QAuO2o

Try to make something similar, export it, and if it still fails to import, send or upload exported *.json files so I can take a look at them..

**Comment by danouche93** [comment]
Okay, I figured out what caused the problem, it was a object that I removed in the scene. 
Now I succeeded to import the asset in Unreal but you can't see any object in the scene ... I can see that the object have the texture and everything but I can't see it in the scene, it's really weird
[Project 3ds max to UE.zip](https://github.com/NegInfinity/ProjectExodus/files/3134913/Project.3ds.max.to.UE.zip)
Here is the .json file that I used to import to unreal, tell me if it works for you..

**Comment by danouche93** [comment]
Here is the link with all the asset : https://drive.google.com/file/d/1JqHrrZ05vlEPW1AfuBmAysBSntUbFSn6/view?usp=sharing

**Comment by NegInfinity** [comment]
Hi.

Sorry for slow reply.

I've imported the project and the scene seemingly imports without problem, along with terrain.

However, due to the way terrain import works, when you import a scene with terrain, it is not imported into the current scene, and instead a **new scene file**, which is by default within /Import/<ProjectName>/<SceneName> , where <SceneName> can be "unnamed scenes" if you exported it thorugh the "exprot scene" menu.

You should see a warning about that when it happens. It looks like this:
![image](https://user-images.githubusercontent.com/15124172/57308348-e2234380-70ee-11e9-8c9d-5c96a1571e4d.png)

And then when import is finished you get this:
![image](https://user-images.githubusercontent.com/15124172/57308392-f6674080-70ee-11e9-9e99-d5ee94ae1415.png)

Basically, you're supposed to wait for shader compilation to finish and then head to the path you saw. In your case it woudl be here:
![image](https://user-images.githubusercontent.com/15124172/57308939-dedc8780-70ef-11e9-95da-852383fa0c1b.png)
